docs/clusterpool.md

```diff
@@ -0,0 +1,28 @@
+[comment]: # ( Copyright Contributors to the Open Cluster Management project )
+
+# ClusterPools
+
+The CLI has commands to manage clusterpools. It allows to manage multiple clusters that are holding the clusterpool management system. 
+
+These clusters are called here `clusterpoolhost`
+
+## Manage ClusterPoolHosts
+
+A clusterpoolhost can be created using the command `cm create clusterpoolhost|cph <clusterpoolhost_name> <options>`. 
+
+The list of clusterpoolhosts can be retreived by calling the command `cm get clusterpoolhosts|cphs <options>`.
+
+The list of clusterpoolhosts is maintained in the `./kube/known-cphs`.
+
+Once the clusterpoolhost is created, the `./kube/config` is updated with a context pointing to that cluster and the clusterpoolhost is set as the current one.
+
+## Use a clusterpoolhost
+
+`cm use <clusterpoolhost_name>` updates the `./kube/config` to point to that cluster.
+
+## Use a cluster clainm managed by a clusterpoolhost
+
+First, the clusterpoolhost which manage the cluster claim must be the current one. This can be done by using the command `cm use <clusterpoolhost_name>`
+
```

Comment:
In this case, I guess current context in`./kube/config` is updated, and KUBECONFIG environment variable is ignored. We should probably spell this out since `cm use` behaves differently depending on the argument. 

How do we determine whether the argument is a clusterpoolhost_name or a cluster_claim_name? We can't have the same name used as both, I guess?

Reply:
I'm not sure if `cm use` should be overloaded like this. What if you use a clusterpool host to deploy another clusterpool host? And then you want to point your context to the child clusterpool host?

Reply:
1) yes all the kubeconfig for clusterpoolhosts are maintained in the .kube/config and don't use the KUBECONFIG. It is explained here: https://github.com/itdove/cm-cli/blob/a1cb003c53184f6a198b61e7ba26b52233c77e17/docs/clusterpool.md#L21

2) We check if the cph name is part of the cphs list to detect if it is a cph or not.

3) yes we can use the same name for a cphs and a cc ...

Reply:
if we want to remove that restriction (cph_name <> cc_name) we have to split the `use` cmd or have a extra parameter for specifying that we are dealing with a cph.
ie:

```
cm use cph <cph_name>
cm use-cph <cph_name>
cm use <cph_name> --cph
```

Reply:
You're right - `ck` would not work with a ClusterClaim `ck`. `ck use` takes the name of a context and they would both have the same context name. But the name of the ClusterPool host is customizable. 

I think you should track current ClusterPool host separately from current context, maybe in `~/.kube/current-cph` . If I understand correctly, if you have ClusterPool host `cph1` with clusters `a`, `b`, `c`, then to switch between the clusters, you need to run:
```
cm use cph1 && cm use a
cm use cph1 && cm use b
cm use cph2 && cm use b
```

Reply:
no, as a,b,c are from the same cph (cph1) . You have first to point to the cph1 (if not yet done) `cm use cph1`, and then run 

```
cm use a; do you stuff on a
cm use b; do you stuff on b
cm use c; do you stuff on c

```